### DISCUSSION TITLE
Unable to add SPF to domain as CNAME already uses the TXT Value

### DISCUSSION THREAD

**Discussion by biswajitnet** [postalserver/postal#1449] [Help with using Postal]
Hello All, 

I came across a scenario where I am unable to add SPF records to my host because the CNAME already exists with the same txt value. 
in my case: I have a cname going from psrp.domain.com to rp.postal.domain.com

Now When I run a mail-tester scan- I am getting no valid spf found. The issue is I ve an SPF for @ TXT Value. but unable to add SPF on psrp again because cname uses the same name. 
Is there  a solution that I am missing here. I can share the domain name personally. 

![image](https://user-images.githubusercontent.com/67558955/123306069-71de1300-d53e-11eb-86a9-d577793f9998.png)

How to get this fixed. I am just 1 point away on Mail-tester and want to nail it. 
![image](https://user-images.githubusercontent.com/67558955/123306231-a81b9280-d53e-11eb-9c19-eaf3f71b8ee0.png)

Regards,
Biswajit.

**Comment by biswajitnet** [comment]
@willpower232  Can you guide me on this, please!

**Comment by willpower232** [comment]
Unfortunately a CNAME means that all records for that subdomain get passed on so the only resolution for this problem is to use A/AAAA records instead

**Comment by attisan** [comment]
@willpower232 we are having the very same problem - and I somehow fail to see how this problem isn't affecting everyone (?).

Postal is set up to use **postal.a.com** and we added the domain **c.com** for mailing. The documentation / guide / wizard tells to set up a `CNAME` from **psrp.c.com** to point to **rp.postal.a.com**. We added a SPF `TXT` record for **c.com** like `v=spf1 mx include:spf.postal.a.com ~all`.  But when mails are sent, the "Return-Path" is something like

> Return-Path: <gfk7tb@psrp.c.com>

and thus, we would need to add a `TXT` dns record for **psrp.c.com**  - what we can't as it is already used by the `CNAME`.

Would make sense to either highlight this in the documentation or otherwise provide a hint ..?

**Reply by willpower232** [reply]
I believe if you use your terminal to `dig TXT psrp.c.com` you should see the SPF record from `rp.postal.a.com` which is already documented https://docs.postalserver.io/install/dns-configuration#return-path

The CNAME forwards all lookups for any type to where it is pointing to so should be fine.

**Reply by attisan** [reply]
trouble with the `CNAME` way is, google (gmail) results in 

> spf=neutral (google.com: [server-ip6] is neither permitted nor denied by best guess record for domain of gfk7tb@psrp.c.com) smtp.mailfrom=gfk7tb@psrp.c.com;

and test pages like https://dnschecker.org/spf-record-validation.php or https://dmarcly.com/tools/spf-record-checker outright refuse to "see" SPF `TXT` entries for respective **psrp.c.com** domains 🤷‍♂️. So even if this would be a valid pattern for mail-servers, it seems hard to test.

**Reply by willpower232** [reply]
I'm afraid I am unable to reproduce your issue, even with gmail

```
Authentication-Results: mx.google.com;
       dkim=pass header.i=@mydomain.com header.s=postal-xrOULx header.b=EmIVBGOe;
       spf=pass (google.com: domain of szfa4f@psrp.mydomain.com designates my:ip:v6:address as permitted sender) 
```

The links you mentioned were able to locate my SPF records with no problem so I suspect something isn't quite right. Does running `dig txt psrp.c.com` show you the correct TXT record and mention the CNAME?

**Reply by attisan** [reply]
could the trouble stem from the fact that `postal.a.com` in our case is a `CNAME` itself and is actually a bit "deeper" ... more like `postal.vminstance.server.a.com` ..?

**Reply by attisan** [reply]
btw `dig TXT psrp.c.com` results in

> psrp.c.com.      3492    IN      CNAME   rp.postal.a.com.

no "magic" resolution delegation is taking place here

**Reply by willpower232** [reply]
ah yes I've seen this before, trying to access the subdomain of a CNAME doesn't work because the CNAME pushes everything deeper as you say

**Reply by attisan** [reply]
> ah yes I've seen this before, trying to access the subdomain of a CNAME doesn't work because the CNAME pushes everything deeper as you say

yes - seems I fixed it by adding another SPF `TXT` record at `rp.postal.vminstance.server.a.com` 🙈

**Reply by attisan** [reply]
perhaps `rp.postal` and `spf.postal` could be streamlined (combined / replaced) as I don't see a hard reason to enforce those to being separated besides semantics? Could save some trouble 😁. Big shoutout to @willpower232 for your lightning response times 💪.

**Reply by willpower232** [reply]
FWIW I don't think Postal necessarily enforces anything, i.e. it doesn't disable sending if it finds the DNS is set up incorrectly.

Postal is more set up to be an unbranded solution for transactional email so the majority of people can be reassured by the DNS checks built in and the advanced users end up ignoring the DNS errors from Postal and use other tools to be pretty sure their records resolve correctly in the end.
